src/unitxt/metrics.py

```diff
@@ -2247,11 +2248,37 @@ def compute(
             )
         result = max([results.score for results in per_reference_results])
 
-        return {
-            self.main_score: result / 5,
-            # "score_name": self.main_score,
-            # "feedback": result.feedback, # removed since this cannot be tested
-        }
+        return {self.main_score: result / 5}
+
+
+class LlamaIndexFaithfulness(LlamaIndexLLMMetric):
+    """LlamaIndex based metric class for evaluating faithfulness."""
+
+    metric_name_prefix = "faithfulness"
+    reference_field = "contexts"  # metric doesn't require reference answers
```

Comment:
Is this needed? I don't see it being used.

Reply:
As this is a referenceless metric, this is there to prevent failures of `InstanceMetric` where no `"references"` field was provided

Reply:
(https://github.com/IBM/unitxt/blob/1242b022919d43716b3f26d67e3c75f262485900/src/unitxt/metrics.py#L877)

Reply:
Thanks for the quick replies on the CR comments. I see.  But a "references" are always passed in unitxt - even for reference-less metrics.  How did you get a failure?

Reply:
Right, that is a different discussion that is outside the scope of this pull request... So if you want I can remove this

Reply:
Yes. For now, please do (it will avoid confusion).

Reply:
No problem. I will say on a more general note that as unitxt provides the possibility for users to use just specific parts of the library (in this case, maybe I want to use a unitxt metric without using the full unitxt pipeline), these sorts of questions might need more thought (cc @elronbandel)